evalml/data_checks/sparsity_data_check.py

```diff
@@ -0,0 +1,90 @@
+        self.threshold = threshold
+        if threshold < 0 or threshold > 1:
+            raise ValueError("Threshold must be a float between 0 and 1, inclusive.")
+        self.unique_count_threshold = unique_count_threshold
+        if unique_count_threshold < 0 or not isinstance(unique_count_threshold, int):
+            raise ValueError("Unique count threshold must be positive integer.")
+
+    def validate(self, X, y=None):
+        """Calculates what percentage of each column's unique values fail to exceed the count threshold.
+        Arguments:
+            X (ww.DataTable, pd.DataFrame, np.ndarray): Features.
+            y (ww.DataColumn, pd.Series, np.ndarray): Ignored.
+        Returns:
+            dict: dict with a DataCheckWarning if there are any sparse columns.
+        Example:
+            >>> import pandas as pd
+            >>> df = pd.DataFrame({
+            ...    'sparse': [float(x) for x in range(100)],
+            ...    'not_sparse': [float(1) for x in range(100)]
+            ... })
+            >>> sparsity_check = SparsityDataCheck(problem_type="multiclass", threshold=0.5, unique_count_threshold=10)
+            >>> assert sparsity_check.validate(df) == {"errors": [],\
+                                                       "warnings": [{"message": "Input columns (sparse) for multiclass problem type are too sparse.",\
+                                                            "data_check_name": "SparsityDataCheck",\
+                                                            "level": "warning",\
+                                                            "code": "TOO_SPARSE",\
+                                                            "details": {"column": "sparse", 'sparsity_score': 0.0}}]}
+        """
+        messages = {
+            "warnings": [],
+            "errors": []
+        }
+
+        X = _convert_to_woodwork_structure(X)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+
+        res = X.apply(SparsityDataCheck.sparsity_score, count_threshold=self.unique_count_threshold)
+        too_sparse_cols = [col for col in res.index[res < self.threshold]]
+        messages["warnings"].extend([DataCheckWarning(message=warning_too_unique.format(col_name,
+                                                                                        self.problem_type),
+                                                      data_check_name=self.name,
+                                                      message_code=DataCheckMessageCode.TOO_SPARSE,
+                                                      details={"column": col_name, "sparsity_score": res.loc[col_name]}).to_dict()
+                                     for col_name in too_sparse_cols])
+        return messages
+
+    @staticmethod
+    def sparsity_score(col, count_threshold=10):
+        """This function calculates a sparsity score for the given value counts by calculating the percentage of
```

Comment:
Description doesn't seem to match what it does. In the test case where the values are all the same, ie `[1, 1, 1, 1, 1, 1, 1, 1]`, we get a `sparsity_score` of 1, but that doesn't mean that 100% of values failed to exceed the count threshold, right? I think it's reversed.

Reply:
You're write, the doc string is just wrong.  Sorry, I feel like I've been going back and forth on this "score."  Let me fix it.